src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -559,28 +591,104 @@ Namespace Microsoft.CodeAnalysis.Semantics
+        ''' is actually a delegate creation.
+        ''' </summary>
+        Private Function CreateConversionOperation(operand As BoundNode, conversionKind As ConversionKind, conversionSyntax As SyntaxNode
+                                                   ) As (methodSymbol As MethodSymbol, operation As Lazy(Of IOperation), isAddressOfDelegateCreation As Boolean)
+            If (conversionKind And VisualBasic.ConversionKind.UserDefined) = VisualBasic.ConversionKind.UserDefined Then
+                Dim userDefinedConversion As BoundUserDefinedConversion = DirectCast(operand, BoundUserDefinedConversion)
+                Return (userDefinedConversion.Call.Method, New Lazy(Of IOperation)(Function() Create(userDefinedConversion.Operand)), isAddressOfDelegateCreation:=False)
+            ElseIf IsDelegateCreationSyntax(conversionKind, conversionSyntax, operand.Syntax) Then
+                ' In this scenario, we're a delegate creation expression involving an AddressOf.
+                Dim methodSymbol As MethodSymbol = Nothing
+                Dim operandLazy As Lazy(Of IOperation)
+                If operand.Kind = BoundKind.DelegateCreationExpression Then
+                    operandLazy = New Lazy(Of IOperation)(Function() CreateBoundDelegateCreationExpressionChildOperation(DirectCast(operand, BoundDelegateCreationExpression)))
+                Else
+                    ' This is an error scenario in which we have a delegate conversion, but it failed to bind correctly, so the child here
+                    ' is either a bad expression or a method group. Delegate to standard operation handling for this.
+                    Debug.Assert(operand.Kind = BoundKind.MethodGroup OrElse
+                                 operand.Kind = BoundKind.BadExpression OrElse
+                                 operand.Kind = BoundKind.AddressOfOperator)
+                    operandLazy = New Lazy(Of IOperation)(Function() Create(operand))
+                End If
+                Return (methodSymbol, operandLazy, isAddressOfDelegateCreation:=True)
+            Else
+                Dim methodSymbol As MethodSymbol = Nothing
+                Return (methodSymbol, New Lazy(Of IOperation)(Function() Create(operand)), isAddressOfDelegateCreation:=False)
+            End If
+        End Function
+
+        Private Function CreateBoundDelegateCreationExpressionOperation(boundDelegateCreationExpression As BoundDelegateCreationExpression) As IDelegateCreationExpression
+            Dim syntax As SyntaxNode = boundDelegateCreationExpression.Syntax
+            Dim type As ITypeSymbol = boundDelegateCreationExpression.Type
+            Dim constantValue As [Optional](Of Object) = ConvertToOptional(boundDelegateCreationExpression.ConstantValueOpt)
+            Dim isImplicit As Boolean = boundDelegateCreationExpression.WasCompilerGenerated OrElse boundDelegateCreationExpression.Syntax.Kind() = SyntaxKind.AddressOfExpression
```

Comment:
>OrElse boundDelegateCreationExpression.Syntax.Kind() = SyntaxKind.AddressOfExpression [](start = 93, length = 85)

Given that tests do not exposed value for IsImplicit, it is not clear what is the impact of this condition. It would also be good to add a clarifying comment. We should really consider an alternative to modify Binder to mark the node as compiler generated instead.

Reply:
Ok, given how the CreateBoundDelegateCreationExpressionChildOperation is implemented, it feels like we should always mark this node as Implicit. The reason is - the operand is going to be using the same syntax node and only one node should be considered explicit for the given syntax node.

---
In reply to: [142809527](https://github.com/dotnet/roslyn/pull/22056#discussion_r142809527) [](ancestors = 142809527)

Reply:
Done, and added a comment.

---
In reply to: [142820740](https://github.com/dotnet/roslyn/pull/22056#discussion_r142820740) [](ancestors = 142820740,142809527)